### DISCUSSION TITLE
Version 2.0.0

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#28] [Announcements]
**Bugfixes:**

1. Fixed DictKeyError inheriting from the wrong parent class.
2. Added extra checks to prevent wrongly configured PGMs and utility keys to break the integration.
3. Fixed where DeviceType would break integration if not set in the Olarm App.

**Features added:**

1. Ability to change and configure the integration without the need to redo the setup. (Note that you will probably need to restart or reload the integration afterwards for the new settings to take effect.

<hr /><em>This discussion was created from the release <a href='https://github.com/rainepretorius/olarm-ha-integration/releases/tag/version2.0.0'>Version 2.0.0</a>.</em>

**Comment by rainepretorius** [comment]
If you get a key error: 'olarm_alarm_code',
You need to redo your config entry as your cnfig entry is of an invalid version.

**Comment by jonathankretzmer** [comment]
Appreciate the new feature :)
